src/plugins/intel_gna/src/common/avx_support.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+// MSVC allows use of AVX2 instructions even if __AVX2__ macro isn't defined
+#if defined(_MSC_VER)
+#    include <intrin.h>
+#    define HAVE_AVX2
+#elif defined(__AVX2__)
```

Comment:
on non _MSC_VER, what's the relation between __AVX2__ flag and cmake's ENABLE_AVX2? Especially, isn't the later implicitly enabled based on existence of the former? If so, then what if one will explicitly decide to ENABLE_AVX2=OFF on __AVX2__ enabled compiler - will the code under the HAVE_AVX2 flag slip in?

Reply:
```__AVX2__``` built-in compiler marco and cmake's ```ENABLE_AVX2``` relation not observed.
For g++: instead, ```__AVX2__``` flag becomes present for compilation with ```-march=core-avx2```. The later does not seem to be set, which may mean ```__AVX2__``` will never appear to be on. please verify.